azure-core fails tests on darwin

### Describe the bug

azure-core is failing to build on my local darwin (x86_64) installation, with 8 tests consistently failing. There's a lot of log output, but here's the tail of it.
```
=========================== short test summary info ============================
FAILED tests/test_rest_polling.py::test_put_operation_location_polling_fail
FAILED tests/test_rest_polling.py::test_put_location_polling_fail - azure.cor...
FAILED tests/test_rest_polling.py::test_patch_operation_location_polling_fail
FAILED tests/test_rest_polling.py::test_patch_location_polling_fail - azure.c...
FAILED tests/async_tests/test_rest_polling_async.py::test_put_operation_location_polling_fail
FAILED tests/async_tests/test_rest_polling_async.py::test_put_location_polling_fail
FAILED tests/async_tests/test_rest_polling_async.py::test_patch_operation_location_polling_fail
FAILED tests/async_tests/test_rest_polling_async.py::test_patch_location_polling_fail
========== 8 failed, 223 passed, 838 deselected, 7 warnings in 12.82s ==========
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x105563ac0>
Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x105a2c100>, 13.31424405)]']
connector: <aiohttp.connector.TCPConnector object at 0x1055637f0>
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x1058da2e0>
Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x1059f7c40>, 13.39427915)]']
connector: <aiohttp.connector.TCPConnector object at 0x1058da310>
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x10561fbe0>
Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x105a2c8e0>, 13.433029846)]']
connector: <aiohttp.connector.TCPConnector object at 0x10589d160>
```
[(full logs)](https://github.com/NixOS/nixpkgs/files/8220609/log.txt)


### Steps To Reproduce

```nix build nixpkgs#python39Packages.azure-core``` on darwin (x86_64) (or against commit `d8cbebf2e95da44192e43552351c1ac65653392f`) (also fails in python310Packages)

### Notify maintainers

@jonringer 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-darwin"`
 - host os: `Darwin 21.3.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.6.0`
 - channels(root): `"nixpkgs-22.05pre346086.e5a50e8f299"`
 - nixpkgs: `/nix/store/30v8bmg615qsdbf2rq9ly53ryw0axjx0-source`
```

--- a/pkgs/development/python-modules/httplib2/default.nix
+++ b/pkgs/development/python-modules/httplib2/default.nix
@@ -55,6 +55,7 @@ buildPythonPackage rec {
   ] ++ lib.optionals (stdenv.isDarwin) [
     # fails with HTTP 408 Request Timeout, instead of expected 200 OK
     "test_timeout_subsequent"
+    "test_connection_close"
   ];
 
   pytestFlagsArray = [

